Add HAProxy helper docs (#7279)
wazuh/wazuh-documentation: source/user-manual/manager/configuring-cluster/advanced-settings.rst

Reviewer: Please use three-spaces indentation. Please use auto-numbering with `#. ` rather than `1.` For example:

```
.. note::

   Text of the note.

.. tabs::

   .. group-tab:: First tab

      #. Install

         .. code-block:: console

            # apt install

      #. Check the installation
```

To enable copying the command to the clipboard in one click, please use output class for command outputs. For example:

```
.. code-block:: none
   :class: output
```

Please use `-  ` for bullet points in lists.

Once installed, regardless of the method, it is needed to do some configurations.

Configuration
-------------

    The HAProxy configuration, present in the ``<HAPROXY_INSTALLATION_PATH>/haproxy.cfg`` must at least have some sections covered:

    - Wazuh agents registration front and backend:

    .. code-block:: console

        frontend wazuh_register
            mode tcp
            bind :1515
            default_backend wazuh_register

        backend wazuh_register
            mode tcp
            balance leastconn
            server master_register <WAZUH_REGISTRY_HOST>:1515 check

    - Configured PID file and stats socket (with ``level admin``):

    .. code-block:: console

        global
            pidfile     /var/run/haproxy.pid
            stats socket /var/lib/haproxy/stats level admin

    - Accessible stats page (optional):

    .. code-block:: console

        frontend stats
            bind *:9000
            stats enable
            stats uri /stats
            stats refresh 5s
            stats admin if TRUE
            option httplog

> Also, it will be needed an init script to manage the HAProxy service. Below we provide an example that can be used.

Author: It is relevant here because it is needed to get the proxy running.

    .. note::

        We recommend using the **load balancer** option. This way, the agents will be able to report to the nodes in a distributed way and it will be the load balancer who assigns which worker they report to. Using this option, we can better distribute the load, and in case of a fall in some worker node, its agents will **reconnect** to another one.

.. _haproxy_installation:

HAProxy Installation
====================

As a load balancer, we recommend using `HAProxy <https://www.haproxy.org/>`_.

There are two main ways to install HAProxy, using `packages <https://github.com/haproxy/wiki/wiki/Packages>`_ or docker `images <https://hub.docker.com/_/haproxy/tags>`_.

Once installed, regardless of the method, it requires some configuration changes.

Configuration
-------------

    The HAProxy configuration, present in the ``<HAPROXY_INSTALLATION_PATH>/haproxy.cfg`` must have some sections covered:

    - Wazuh agents frontend and backend registration:

>         The *backend* is a set of servers(Wazuh cluster nodes) that receive the forwarded agent connections and can be defined by:

Reviewer 2: ```suggestion
        The *backend* is a set of servers (Wazuh cluster nodes) that receive the forwarded agent connections, and is defined by:
```

            - the type of load balancing

>             - which load balance algorithm to use

Author: Rephrased

                server master_register <WAZUH_REGISTRY_HOST>:1515 check

        A *frontend* defines how requests should be forwarded to backends and is composed of:

            - the type of load balancing
            - the port to bind the connections
            - the default backend to forward requests

        .. code-block:: console

            frontend wazuh_register
                mode tcp
                bind :1515
                default_backend wazuh_register

    - Configured PID file and stats socket (with ``level admin``):

    .. code-block:: console

        global
            pidfile     /var/run/haproxy.pid
            stats socket /var/lib/haproxy/stats level admin

    As an optional step, the stats page can be enabled with:

    .. code-block:: console

        frontend stats
            bind *:9000
            stats enable
            stats uri /stats
            stats refresh 5s
            stats admin if TRUE
            option httplog

Lifecycle script handler
------------------------

To manage the HAProxy service, we will need a script that is in charge of starting and stopping it. Below we provide a working example.

> .. code-block:: bash

Author: I decided to use raw html to put it into `<details>` tag.

.. Copyright (C) 2015, Wazuh, Inc.

.. meta::
  :description: Learn more about how to deploy a Wazuh cluster. In this section of our documentation, we explain more about the agents connections.

.. _advanced_cluster_settings:

*****************
Advanced Settings
*****************

    In this section, we are going to explore other ways to configure our cluster. We'll see that besides the basic way described in our :ref:`Getting started <gt-cluster>`, we have more options to use:

    - HAProxy as a balancer
    - HAProxy as a balancer + HAProxy helper as a real-time configuration manager.
    - Agents directly connected to managers using failover mode

.. _haproxy_installation:

> HAProxy Installation
> ====================

Reviewer: ```suggestion
Installation
^^^^^^^^^^^^
```

> We recommend using the **load balancer** option. This way, the agents will be able to report to the nodes in a distributed way and it will be the load balancer who assigns which worker they report to.
> Using this option, we can better distribute the load, and in case of a fall in some worker node, its agents will **reconnect** to another one.
> 
> As a load balancer, we suggest using `HAProxy <https://www.haproxy.org/>`_.
> 
> There are two main ways to install HAProxy, using packages or Docker images.

Reviewer: ```suggestion
Using a load balancer, such as `HAProxy <https://www.haproxy.org/>`__, ensures the Wazuh agents register and report to Wazuh manager nodes in a distributed way. The load balancer assigns manager nodes to the agents improving load distribution. If a Wazuh manager node fails, the Wazuh agents reconnect to another node.

There are two main ways to install HAProxy.

-  Using system and PPA packages
-  Using Docker images
```

    This is the basic configuration:

    .. code-block:: yaml

        dataplaneapi:
            host: 0.0.0.0
            port: 5555
            transaction:
                transaction_dir: /tmp/haproxy
            user:
            - insecure: true
              password: <DATAPLANE_PASSWORD>
              name: <DATAPLANE_USER>
        haproxy:
            config_file: /etc/haproxy/haproxy.cfg
            haproxy_bin: /usr/sbin/haproxy
            reload:
                reload_delay: 5
                reload_cmd: service haproxy reload
                restart_cmd: service haproxy restart

    .. note::

        Is needed to replace ``<DATAPLANE_USER>`` and ``<DATAPLANE_PASSWORD>`` with the chosen user and password.

    To enable it will depend on the :ref:`installation method <haproxy_installation>`.

    .. warning::

        For the correct operation of the helper, there must not be a frontend, with the port **1514**, in the ``haproxy.cfg`` file beforehand.

    .. tabs::
        .. group-tab:: Packages

            1. Download the binary file for the installed HAProxy version

            .. code-block:: console

>                 # curl -sL https://github.com/haproxytech/dataplaneapi/releases/download/v2.8.X/dataplaneapi_2.8.X_linux_x86_64.tar.gz | tar xz && cp dataplaneapi /usr/local/bin/

Reviewer 3: ```suggestion
                # curl -sL https://github.com/haproxytech/dataplaneapi/releases/download/v2.8.7/dataplaneapi_2.8.7_linux_x86_64.tar.gz | tar xz && cp dataplaneapi /usr/local/bin/
```
One last comment on my side: are these Xs intentional? If we don't want to include the patch in case they update it (which seems fine to me), we should at least tell the user where they can find the latest one: 
- https://github.com/haproxytech/dataplaneapi/releases/

Maybe something like 
```
1. Download the binary file for the installed HAProxy version. You can find the available versions [here](https://github.com/haproxytech/dataplaneapi/releases/). 
```

                case "$ret" in
                0)
                    echo "${BASENAME} is running."
                    ;;
                1)
                    echo "${BASENAME} dead, but $PIDFILE exists."
                    ;;
                *)
                    echo "${BASENAME} not running."
                    ;;
                esac
                exit $ret
                ;;
            *)
                echo "Usage: /etc/init.d/${BASENAME} {start|stop|reload|restart|status}"
                exit 2
                ;;
            esac

            :

        .. raw:: html

            </details>

        And a :ref:`Configuration file <haproxy_configuration>` to get the service up and running.

        1. It will be needed to put these files in the same directory and build the image

        .. code-block:: console

            # tree
            .
            ├── Dockerfile
            ├── entrypoint.sh
            ├── haproxy.cfg
            └── haproxy-service

        .. code-block:: console

            # docker build --tag=haproxy-deploy .

        2. After building the image can we run the haproxy service

        .. code-block:: console

>             # docker run haproxy-deploy
>             TCPLOG: true HTTPLOG: true
>             * Starting haproxy haproxy                                                                                                                                                                                          [NOTICE]   (33) : haproxy version is 2.8.9-1842fd0
>             [NOTICE]   (33) : path to executable is /usr/sbin/haproxy
>             [ALERT]    (33) : config : parsing [/etc/haproxy/haproxy.cfg:3] : 'pidfile' already specified. Continuing.

Reviewer 3: This was misaligned:
```suggestion
            # docker run haproxy-deploy
            TCPLOG: true HTTPLOG: true
            * Starting haproxy haproxy
            [NOTICE]   (33) : haproxy version is 2.8.9-1842fd0
            [NOTICE]   (33) : path to executable is /usr/sbin/haproxy
            [ALERT]    (33) : config : parsing [/etc/haproxy/haproxy.cfg:3] : 'pidfile' already specified. Continuing.
```

![image](https://github.com/wazuh/wazuh-documentation/assets/23361101/24d1856a-a2cf-46ce-a13a-58136ef5175b)

            global
                chroot      /var/lib/haproxy
                pidfile     /var/run/haproxy.pid
                maxconn     4000
                user        haproxy
                group       haproxy
                stats socket /var/lib/haproxy/stats level admin
                log 127.0.0.1 local2 info

            defaults
                mode http
                maxconn 4000
                log global
                option redispatch
                option dontlognull
                option tcplog
                timeout check 10s
                timeout connect 10s
                timeout client 1m
                timeout queue 1m
                timeout server 1m
                retries 3

            frontend wazuh_register
                mode tcp
                bind :1515
                default_backend wazuh_register

            backend wazuh_register
                mode tcp
                balance leastconn
                server master <IP_OR_DNS_OF_WAZUH_MASTER_NODE>:1515 check
                server worker1 <IP_OR_DNS_OF_WAZUH_WORKER_NODE>:1515 check
                server workern <IP_OR_DNS_OF_WAZUH_WORKER_NODE>:1515 check

> 

Reviewer 3: The current configuration is fine and will allow agents to register, but once registered (on port 1515) they won't be able to connect to any manager since it happens on port 1514. An additional frontend and backend is required for this, but only if the HAProxy helper is not used. If the helper is used, it will add the `wazuh_reporting` frontend and backend (the name will depend on the `<haproxy_backend>` option):

```suggestion
            frontend wazuh_reporting_front
                mode tcp
                bind :1514 name wazuh_cluster_front_bind
                default_backend wazuh_cluster

            backend wazuh_reporting
                mode tcp
                balance leastconn
                server master <IP_OR_DNS_OF_WAZUH_MASTER_NODE>:1514 check
                server worker1 <IP_OR_DNS_OF_WAZUH_WORKER_NODE>:1514 check
                server worker2 <IP_OR_DNS_OF_WAZUH_WORKER_NODE>:1514 check

```

It might be useful to highlight the lines and explain to the user that this part should not be included if they intend to use the helper. 

Or maybe the other way around. Do not add it to `haproxy.cfg` and explain/share the block that should be included if the helper is not used.

        The provided setup is ready to work with a Wazuh cluster. Some of the sections covered are:

            - The *backend* is a set of servers (Wazuh cluster nodes) that receive the forwarded agent connections, and is defined by:

                - the load balancing mode
                - which load balance algorithm to use
                - a list of servers and ports, in this case, the default one pointing to the master node of the cluster.

            .. code-block:: console

                backend wazuh_register
                    mode tcp
                    balance leastconn
                    server master_node <WAZUH_REGISTRY_HOST>:1515 check

            - A *frontend* defines how requests should be forwarded to backends and is composed of:

                - the type of load balancing
                - the port to bind the connections
                - the default backend to forward requests

            .. code-block:: console

                frontend wazuh_register
                    mode tcp
                    bind :1515
                    default_backend wazuh_register

    2. Now can we start the service with

        .. code-block:: console

>             # service haproxy start
>             * Starting haproxy haproxy                                                                                                                                                                                          [NOTICE]   (13231) : haproxy version is 2.8.9-1ppa1~jammy
>             [NOTICE]   (13231) : path to executable is /usr/sbin/haproxy
>             [ALERT]    (13231) : config : parsing [/etc/haproxy/haproxy.cfg:3] : 'pidfile' already specified. Continuing.

Reviewer 3: ```suggestion
            # service haproxy start
            * Starting haproxy haproxy                                                                                                                                                                                          
            [NOTICE]   (13231) : haproxy version is 2.8.9-1ppa1~jammy
            [NOTICE]   (13231) : path to executable is /usr/sbin/haproxy
            [ALERT]    (13231) : config : parsing [/etc/haproxy/haproxy.cfg:3] : 'pidfile' already specified. Continuing.
```

    :title: HAProxy helper flow
    :alt: HAProxy helper flow
    :align: center
    :width: 80%

How to enable it
----------------

.. note::
    The recommended version of HAProxy is the 2.8 LTS.

To use this feature is required to have a :ref:`HAProxy <haproxy_installation>` instance balancing the cluster using the **least connections** algorithm.

Dataplane API configuration
^^^^^^^^^^^^^^^^^^^^^^^^^^^

    The Dataplane API is used by the helper to communicate with HAProxy and update the configuration according to the changes in the Wazuh cluster.

    This is the basic configuration:

    .. code-block:: yaml

        dataplaneapi:
            host: 0.0.0.0
            port: 5555
            transaction:
                transaction_dir: /tmp/haproxy
            user:
            - insecure: true
              password: <DATAPLANE_PASSWORD>
              name: <DATAPLANE_USER>
        haproxy:
            config_file: /etc/haproxy/haproxy.cfg
            haproxy_bin: /usr/sbin/haproxy
            reload:
                reload_delay: 5
                reload_cmd: service haproxy reload
                restart_cmd: service haproxy restart

>     To enable it will depend on the :ref:`installation method <haproxy_installation>`.

Reviewer 3: I think reminding the user to modify `<DATAPLANE_PASSWORD>` and `<DATAPLANE_USER>` can be helpful.

    .. tabs::
        .. group-tab:: Packages

            1. Download the binary file for the installed HAProxy version

            .. code-block:: console

>                 # curl -sL https://github.com/haproxytech/dataplaneapi/releases/download/v2.8.X/dataplaneapi_2.8.X_linux_x86_64.tar.gz | tar xz | cp dataplaneapi /usr/local/bin/

Reviewer 3: The url contains `.X`, I guess those should be replaced. Also, cp must be done in a different command, not using a pipe:
```suggestion
                # curl -sL https://github.com/haproxytech/dataplaneapi/releases/download/v2.8.7/dataplaneapi_2.8.7_linux_x86_64.tar.gz | tar xz && cp dataplaneapi /usr/local/bin/
```

            1. Put the configuration into ``dataplaneapi.yaml``

                .. code-block:: console

                    # tree
                    .
                    ├── dataplaneapi.yml
                    ├── Dockerfile
                    ├── entrypoint.sh
                    ├── haproxy.cfg
                    └── haproxy-service

            2. Modify the Dockerfile to include ``dataplaneapi.yaml`` during the build

                .. code-block:: dockerfile
                    :emphasize-lines: 4

                    FROM haproxytech/haproxy-ubuntu:2.8

                    COPY haproxy.cfg /etc/haproxy/haproxy.cfg
                    COPY dataplaneapi.yml /etc/haproxy/dataplaneapi.yml
                    COPY haproxy-service /etc/init.d/haproxy
                    COPY entrypoint.sh /entrypoint.sh

                    RUN chmod +x /etc/init.d/haproxy
                    RUN chmod +x /entrypoint.sh

                    ENTRYPOINT [ "/entrypoint.sh" ]

            3. Modify the ``entrypoint.sh`` to start the dataplaneapi process

                .. code-block:: bash
                    :emphasize-lines: 27

                    #!/usr/bin/env bash

>                     tcplog_enabled=$(echo "${TCPLOG:-true}" | tr '[:upper:]' '[:lower:]')
>                     httplog_enabled=$(echo "${HTTPLOG:-true}" | tr '[:upper:]' '[:lower:]')
>                     echo TCPLOG: $tcplog_enabled HTTPLOG: $httplog_enabled
>                     if [ "$tcplog_enabled" = "true" ]; then
>                     sed -i 's/# option tcplog/  option tcplog/' /etc/haproxy/haproxy.cfg
>                     else
>                     sed -i 's/  option tcplog/# option tcplog/' /etc/haproxy/haproxy.cfg
>                     fi
>                     if [ "$httplog_enabled" = "true" ]; then
>                     sed -i 's/# option httplog/  option httplog/' /etc/haproxy/haproxy.cfg
>                     else
>                     sed -i 's/  option httplog/# option httplog/' /etc/haproxy/haproxy.cfg
>                     fi
>                     if [ "$tcplog_enabled" = "true" ] || [ "$httplog_enabled" = "true" ]; then
>                     sed -i 's/#  log /  log /' /etc/haproxy/haproxy.cfg
>                     else
>                     sed -i 's/  log /#  log /' /etc/haproxy/haproxy.cfg
>                     fi
> 
>                     # Set env-file
>                     env > /.env-file

Reviewer 3: Same as before:
```suggestion
```

                    # Start HAProxy service
                    service haproxy start
                    # Start HAProxy Data Plane API
                    dataplaneapi -f /etc/haproxy/dataplaneapi.yml &

                    tail -f /dev/null

            4. Build and run the image

                .. code-block:: console

                    # docker build --tag=haproxy-deploy .

                .. code-block:: console

>                     # docker run -p 5555:5555 haproxy-deploy
>                     TCPLOG: true HTTPLOG: true
>                     * Starting haproxy haproxy                                                                                                                                                                                          [NOTICE]   (33) : haproxy version is 2.8.9-1842fd0
>                     [NOTICE]   (33) : path to executable is /usr/sbin/haproxy
>                     [ALERT]    (33) : config : parsing [/etc/haproxy/haproxy.cfg:3] : 'pidfile' already specified. Continuing.

Reviewer 3: ```suggestion
                    # docker run -p 5555:5555 haproxy-deploy
                    TCPLOG: true HTTPLOG: true
                    * Starting haproxy haproxy
                    [NOTICE]   (33) : haproxy version is 2.8.9-1842fd0
                    [NOTICE]   (33) : path to executable is /usr/sbin/haproxy
                    [ALERT]    (33) : config : parsing [/etc/haproxy/haproxy.cfg:3] : 'pidfile' already specified. Continuing.
```

            5. Verify the API is running properly

                .. code-block:: console

                    # curl -X GET --user haproxy:haproxy http://localhost:5555/v2/info
                    {"api":{"build_date":"2024-05-13T14:06:03.000Z","version":"v2.9.3 59f34ea1"},"system":{}}

On the Wazuh's side, we will include the ``<haproxy_helper>...</haproxy_helper>`` labels in the :ref:`configuration <haproxy_helper>` file (``/var/ossec/etc/ossec.conf``)  within the ``<cluster>...</cluster>`` section.

> We are going to configure a basic HAProxy helper within an already configured cluster master node:

Reviewer 3: We should emphasize that this configuration is only necessary on the master node.

    - :ref:`haproxy_disabled <haproxy_disabled>`: Indicates whether the helper will be enabled or not in the master node.
    - :ref:`haproxy_address <haproxy_address>`: Address (IP or DNS) to connect with HAProxy.
    - :ref:`haproxy_user <haproxy_user>`: Username to authenticate with HAProxy.
    - :ref:`haproxy_password <haproxy_password>`: Password to authenticate with HAProxy.

.. code-block:: xml

    <cluster>
        <name>wazuh</name>
        <node_name>master-node</node_name>
        <key>c98b62a9b6169ac5f67dae55ae4a9088</key>
        <node_type>master</node_type>
        <port>1516</port>
        <bind_addr>0.0.0.0</bind_addr>
        <nodes>
            <node>WAZUH-MASTER-ADDRESS</node>
        </nodes>
        <hidden>no</hidden>
        <disabled>no</disabled>

>         <haproxy_helper>
>             <haproxy_disabled>no</haproxy_disabled>
>             <haproxy_address><HAPROXY_ADDRESS></haproxy_address>
>             <haproxy_user><DATAPLANE_USER></haproxy_user>
>             <haproxy_password><DATAPLANE_PASSWORD></haproxy_password>
>       </haproxy_helper>

Reviewer 3: ```suggestion
        <haproxy_helper>
            <haproxy_disabled>no</haproxy_disabled>
            <haproxy_address><HAPROXY_ADDRESS></haproxy_address>
            <haproxy_user><DATAPLANE_USER></haproxy_user>
            <haproxy_password><DATAPLANE_PASSWORD></haproxy_password>
        </haproxy_helper>
```

.. _haproxy_helper_setup:

HAProxy helper
==============

This is an optional tool to manage HAProxy configuration depending on the Wazuh cluster status in real-time.
Provides the manager with the ability to automatically balance the agent TCP sessions.

Some of its key features are:

* Add/remove new servers to the Wazuh backend (1514/tcp) when detecting changes on the Wazuh cluster (e.g. new workers connected).
* Balance excess agents per node when adding new servers to the Wazuh backend.
* Balance agents when detecting an imbalance that exceeds the given tolerance.

.. thumbnail:: /images/manual/cluster/haproxy-helper-architecture.png
    :title: HAProxy helper architecture
    :alt: HAProxy helper architecture
    :align: center
    :width: 80%

The helper runs in an independent thread, that initiates with the ``wazuh-cluster`` daemon, and completes the next flow:

.. thumbnail:: /images/manual/cluster/haproxy-helper-flow.png
    :title: HAProxy helper flow
    :alt: HAProxy helper flow
    :align: center
    :width: 80%

How to enable it
----------------

.. note::

>     The recommended version of HAProxy, that we currently test, is the 2.8 LTS.

Reviewer 2: ```suggestion
    The recommended version of HAProxy is the 2.8 LTS.
```

> To use this feature is needed to have a :ref:`HAProxy <haproxy_installation>`  instance balancing the cluster using the **least connections** algorithm.

Reviewer 2: ```suggestion
To use this feature it is required to have a :ref:`HAProxy <haproxy_installation>`  instance balancing the cluster using the **least connections** algorithm.
```

    1. This is the basic configuration (``<HAPROXY_INSTALLATION_PATH/dataplaneapi.yml>``) to enable it:

    .. code-block:: yaml

        dataplaneapi:
            host: 0.0.0.0
            port: 5555
            transaction:
                transaction_dir: /tmp/haproxy
            user:
            - insecure: true
                password: <DATAPLANE_PASSWORD>
                name: <DATAPLANE_USER>
        haproxy:
            config_file: /etc/haproxy/conf.d/haproxy.cfg
            haproxy_bin: /usr/sbin/haproxy
            reload:
                reload_delay: 5
                reload_cmd: service haproxy reload
                restart_cmd: service haproxy restart

    2. Start the process with:

    .. code-block:: console

        dataplaneapi -f <HAPROXY_INSTALLATION_PATH/dataplaneapi.yml>

    3. Add Dataplane API credentials in ``<HAPROXY_INSTALLATION_PATH>/haproxy.cfg``:

    .. code-block:: console

        userlist haproxy-dataplaneapi
            user <DATAPLANE_USER> insecure-password <DATAPLANE_PASSWORD>

    4. Restart the HAProxy service:

    .. code-block:: console

        service haproxy restart

> In the Wazuh side, the :ref:`configuration <haproxy_helper>` file (``/var/ossec/etc/ossec.conf``) we will include the ``<haproxy_helper>...</haproxy_helper>`` labels within the ``<cluster>...</cluster>`` section.

Reviewer 2: ```suggestion
On Wazuh's side, we will include the ``<haproxy_helper>...</haproxy_helper>`` labels in the :ref:`configuration <haproxy_helper>` file (``/var/ossec/etc/ossec.conf``) within the ``<cluster>...</cluster>`` section.
```

* Balance excess agents per node when adding new servers to the Wazuh backend.
* Balance agents when detecting an imbalance that exceeds the given tolerance.

.. thumbnail:: /images/manual/cluster/haproxy-helper-architecture.png
    :title: HAProxy helper architecture
    :alt: HAProxy helper architecture
    :align: center
    :width: 80%

The helper runs in an independent thread, that initiates with the ``wazuh-cluster`` daemon, and completes the next flow:

.. thumbnail:: /images/manual/cluster/haproxy-helper-flow.png
    :title: HAProxy helper flow
    :alt: HAProxy helper flow
    :align: center
    :width: 80%

How to enable it
----------------

.. note::
    The recommended version of HAProxy is the 2.8 LTS.

To use this feature is required to have a :ref:`HAProxy <haproxy_installation>` instance balancing the cluster using the **least connections** algorithm.

Dataplane API configuration
^^^^^^^^^^^^^^^^^^^^^^^^^^^

    The Dataplane API is used by the helper to communicate with HAProxy and update the configuration according to the changes in the Wazuh cluster.

>     1. This is the basic configuration (``<HAPROXY_INSTALLATION_PATH/dataplaneapi.yml>``) to enable it:

Reviewer 3: ```suggestion
    1. This is the basic configuration (``<HAPROXY_INSTALLATION_PATH>/dataplaneapi.yml``) to enable it:
```
